hilti/include/rt/types/vector.h

```diff
@@ -195,46 +349,45 @@ namespace detail::adl {
 template<typename T, typename Allocator>
 inline std::string to_string(const Vector<T, Allocator>& x, adl::tag /*unused*/) {
     using detail::adl::to_string;
-    return fmt("[%s]",
-               rt::join(rt::transform(x, [](const std::optional<T>& y) { return (y ? rt::to_string(*y) : "(unset)"); }),
-                        ", "));
+    return fmt("[%s]", rt::join(rt::transform(x, [](const T& y) { return rt::to_string(y); }), ", "));
 }
 
 template<typename T, typename Allocator>
 inline std::string to_string(const std::vector<T, Allocator>& x, adl::tag /*unused*/) {
-    return to_string(static_cast<const Vector<T, Allocator>&>(x), adl::tag{});
+    using detail::adl::to_string;
+    return fmt("[%s]", rt::join(rt::transform(x, [](const T& y) { return rt::to_string(y); }), ", "));
 }
 
 inline std::string to_string(const vector::Empty& /* x */, adl::tag /*unused*/) { return "[]"; }
 
 template<typename T, typename Allocator>
-inline std::string to_string(const vector::SafeIterator<T, Allocator>& /*unused*/, adl::tag /*unused*/) {
+inline std::string to_string(const vector::Iterator<T, Allocator>& /*unused*/, adl::tag /*unused*/) {
     return "<vector iterator>";
 }
 
 template<typename T, typename Allocator>
-inline std::string to_string(const vector::SafeConstIterator<T, Allocator>& /*unused*/, adl::tag /*unused*/) {
+inline std::string to_string(const vector::ConstIterator<T, Allocator>& /*unused*/, adl::tag /*unused*/) {
     return "<const vector iterator>";
 }
 
 template<typename T, typename Allocator>
 inline auto safe_begin(const Vector<T, Allocator>& x, adl::tag /*unused*/) {
```

Comment:
Do we need the `safe_*()` versions still just because the code gen still generates them>

Reply:
Not for `rt::Vector`, but I'd say let's leave them in for now until we have all containers cleaned up. After that we can remove it, and do not need to introduce workarounds for intrinsically safe containers.

Reply:
Ack.